src/main/kotlin/org/jetbrains/kotlinx/jupyter/messaging/MessageHandlerImpl.kt

```diff
@@ -3,11 +3,11 @@ package org.jetbrains.kotlinx.jupyter.messaging
 import org.jetbrains.kotlinx.jupyter.api.libraries.RawMessage
 import org.jetbrains.kotlinx.jupyter.execution.JupyterExecutor
 import org.jetbrains.kotlinx.jupyter.messaging.comms.CommManagerInternal
-import org.jetbrains.kotlinx.jupyter.repl.ReplForJupyter
+import org.jetbrains.kotlinx.jupyter.repl.impl.ReplForJupyterImpl
 import java.util.concurrent.atomic.AtomicLong
 
 class MessageHandlerImpl(
-    private val repl: ReplForJupyter,
+    private val repl: ReplForJupyterImpl,
```

Comment:
Don't do this, it's a bad path. Just add an extension
```
fun Any.closeIfNeeded() {
    if (this is Closeable) close()
}
```

And use this extension to close the repl here

Reply:
That's indeed nice move, +1